ECIPs/ECIP-1042.md

```diff
@@ -0,0 +1,63 @@
+    ECIP: 1042
+    Title: GASPAY opcode
+    Author: Igor Artamonov <splix@etcdevteam.com>
+    Status: Draft
+    Type: Standard
+    Created: 2018-03-08
+    
+# ECIP-1042 - GASPAY opcode    
+    
+# Abstract
+Add an operation to consume gas and add amount paid for that gas to token balance.
+
+# Motivation
+A useful Smart Contract (i.e. valuable) should expect small reward for its execution. A micropayment could be very 
+small a sub-cent payment but sending an additional value to a contract will make it more complicated and cost of such 
+transfer or contract execution could exceed requested reward.
+
+It’s proposed here to introduce a new EVM Operation that will allow a developer to consume a portion of transaction fee 
+(amount paid for a gas) and transfer it to current Smart Contract. 
+
+Using a gas payment provided with transaction provides flexibility and independence from market price of Ether. Gas Price 
+is dynamic value and supposed to represent fair market price of contract execution, therefore making micropayment value 
+proportional to const of transaction and more stable in long term. 
+
+# Specification
+Provide a new operation `GASPAY a` where `a` is amount of gas to consume, and `0x0 <= a <= 0x0fffff`. The payment for 
+that gas goes to contract address. As a return it puts consumed value (Wei) into the stack.
```

Comment:
I suggest we be more specific on this.

1. If a `DELEGATECALL` happened, does the payment goes to the parent contract (which has the address field set in context), or the delegated contract?
2. What is the case if we're in `CREATE` or a contract creation transaction?

Reply:
1. to delegated contract
2. not sure what you mean